atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/keyvalue/api/watch/LockWatchEventCacheImpl.java

```diff
@@ -100,7 +105,13 @@ public TransactionsLockWatchUpdate getUpdateForTransactions(
         lastKnownVersion.ifPresent(version -> assertTrue(version.version() <= timestampMapping.lastVersion().version(),
```

Comment:
For completeness, can you also remove the "Preconditions.checkArgument(!startTimestamps.isEmpty(), "Cannot get events for empty set of transactions");" here; it's checked in the TimestampMapping

Reply:
Good point - removed